runtime/interpreter/storagemapkey.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package interpreter
+
+import "github.com/onflow/atree"
+
+type StorageMapKey interface {
+	isStorageMapKey()
+	AtreeValue() atree.Value
+	AtreeValueHashInput(v atree.Value, _ []byte) ([]byte, error)
+	AtreeValueCompare(storage atree.SlabStorage, value atree.Value, otherStorable atree.Storable) (bool, error)
```

Comment:
These two functions are really just "static" functions of an implementation of this interface, not instance methods: For a particular type that implements the `StorageMapKey` interface, we want to get the type's hash input function and value comparison functions.

Unfortunately, I didn't quite come up with a better way of expressing/implementing that.